src/NLog/Targets/FileTarget.cs

```diff
@@ -1058,22 +1052,21 @@ private void RollArchivesForward(string fileName, string pattern, int archiveNum
             string newFileName = ReplaceNumberPattern(pattern, archiveNumber);
             RollArchivesForward(newFileName, pattern, archiveNumber + 1);
 
-            var shouldCompress = archiveNumber == 0;
-            try
-            {
-                RollArchiveForward(fileName, newFileName, shouldCompress);
-            }
-            catch (IOException)
```

Comment:
> why has this changed to two method

This is a slight refactoring. See the change in the `RollArchiveForward()` method. This method was previously called to "move archive files forward" (when `shouldCompress` is `false`), but also to archive log files (when `shouldCompress` is `true`). A method must have only one clear responsibility and I want us to make a clear distinction between "rolling forward" and archiving. Therefore I specialized this method to cater for moving archive files forward only. When archiving a log file, we should call `ArchiveFile()` instead.

As a side-effect of this change, the two method calls here make it explicit that we archive the first file, but "roll" all of the rest.